src/hooks/useLogEvent.ts

```diff
@@ -0,0 +1,12 @@
+import { logEvent } from 'firebase/analytics';
+import { analytics } from '@/firebase';
+
+export const useLogEvent = () => {
+  const triggerEvent = <T>(event: string, eventParams?: T) => {
```

Comment:
we can make a `EventType` type e.g. `type EventType = "save_image" | "share_image" ...` to avoid mispells and gradually add more

Reply:
added EventType: 401b674